### DISCUSSION TITLE
CA certificates extracted from Mozilla should be signed

### DISCUSSION THREAD

**Discussion by earlchew** [curl/curl#10333] [Ideas]
#### Proposal

https://curl.se/ca/ include signatures to allow users to check the authenticity of the data.

#### Discussion

Users can download a CA root certificate bundle https://curl.se/docs/sslcerts.html.

While the connection is secure, and the server can be authenticated, it is harder to be assured of the authenticity content of the bundle itself. The site provides SHA-256 checksums, but with this alone users can only check the data integrity of the bundle.

By contrast Debian provides both the bundle and a signed digest at https://ftp.debian.org/debian/pool/main/c/ca-certificates/ca-certificates_20210119.dsc:
```
...
Checksums-Sha1:
 bce5a8fac45456dbebf256f3a812c6cd0a853e3e 239608 ca-certificates_20211016.tar.xz
Checksums-Sha256:
 2ae9b6dc5f40c25d6d7fe55e07b54f12a8967d1955d3b7b2f42ee46266eeef88 239608 ca-certificates_20211016.tar.xz
Files:
 5cce77de047611c4b9384d4ce52d9204 239608 ca-certificates_20211016.tar.xz
Dgit: 07de54fdcc5806bde549e1edf60738c6bccf50e8 debian archive/debian/20211016 https://git.dgit.debian.org/ca-certificates

-----BEGIN PGP SIGNATURE-----

iQJIBAEBCgAyFiEEVXgdqzTmGgnvuIvhnbAjVVb4z60FAmFq+toUHGpjcmlzdGF1
QGRlYmlhbi5vcmcACgkQnbAjVVb4z61tDw//d4SObXDb4Uw13ebuGhTRtJyAg9aH
...
-----END PGP SIGNATURE-----
```

**Comment by dfandrich** [comment]
What is the threat model that you want to protect against? The connection to
the server is secure, so it's not MITM. If an attacker is able to replace the
certificate file then he's also able to replace the signature file. If you
don't already have the correct signing key downloaded, then you won't be able
to tell if that new signature is the genuine one or not. And since the
downloading and signing is being done on the server, the real key would be
there as well and also a target for an attacker. Yes, someone could design an
implement a more secure automatic signing architecture but that's going to be
complicated and expensive and somebody has to take responsibility for
not just doing it but monitoring and auditing it or else you don't know that
it's providing anything more than the appearance of higher security. And is
Mozilla signing its original bundle? How do we know that wasn't compromised?

If you're concerned that the bundle may be compromised, then you should
probably not be trusting haxx.se at all and rather run the conversion script on
your own machine where you can cut out the middle man and take whatever
additional precautions you feel are necessary.

**Reply by bagder** [reply]
I concur with @dfandrich. I don't think adding signatures in our end does much good and would be a lot of work.
